opsdroid/connector/matrix/connector.py

```diff
@@ -216,9 +218,14 @@ async def connect(self):
             self.connection.user_id = self.mxid
 
         elif self.mxid is not None and self.password is not None:
-            login_response = await self.connection.login(
-                password=self.password, device_name=self.device_name
-            )
+            try:
+                login_response = await self.connection.login(
+                    password=self.password, device_name=self.device_name
+                )
+            except Exception:
+                _LOGGER.error("connexion error")
```

Comment:
I think what this needs to be is something like this:
```suggestion
            
            login_response = await self.connection.login(
                password=self.password, device_name=self.device_name
            )
            if isinstance(login_response, LoginError):
                logger.error("Login to the matrix homeserver failed with error: %s", login_response)
```

Reply:
Good to know on the socket error. However, I think that we probably *also* want to catch non-socket errors at login which would be returned as a `LoginError` class? Perhaps a hybrid of both is the answer?

Reply:
In fact, it's what we do:
We surrounded the login() method with a try / except, but we kept the if isinstance(login_response, LoginError) that was initially the code.